src/ergw_http_api.erl

```diff
@@ -0,0 +1,67 @@
+%% Copyright 2017, Travelping GmbH <info@travelping.com>
+
+%% This program is free software; you can redistribute it and/or
+%% modify it under the terms of the GNU General Public License
+%% as published by the Free Software Foundation; either version
+%% 2 of the License, or (at your option) any later version.
+
+-module(ergw_http_api).
+
+-export([init/0]).
+
+-define(DEFAULT_PORT,	8000).
+-define(DEFAULT_IP,     {127, 0, 0, 1}).
+-define(ACCEPTORS_NUM,  100).
+
+init() ->
+    HttpConfig = application:get_env(ergw, http_api),
+    case HttpConfig of
+        undefined ->
+            lager:debug("HTTP API will not be started because of lack of configuration~n"),
+            ok;
+        {ok, HttpOpts0} ->
+            lager:debug("HTTP API listener options: ~p", [HttpOpts0]),
+            try validate_options(HttpOpts0) of
```

Comment:
i would prefer if this where done as part of the main ergw configuration validation

There should something like in ergw_config.erl:
```
validate_option(http_api, Value) when is_list(Value) ->
    validate_options(fun http_api:validate_options/2, Value);
```

The validate_options needs to be adjusted to follow the scheme used in ergw_config